File: pub_sub/src/string_pkg/src/sub_string.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

using namespace ros;

// callback
void callback(const std_msgs::String::ConstPtr &msg)
{
    ROS_INFO("Hello, %s", msg->data.c_str());
}
int main(int argc, char **argv)
{
    // node
    init(argc, argv, "listener", init_options::AnonymousName);
    NodeHandle nh;

    // subscribe
    Subscriber sub = nh.subscribe("string_topic", 1, callback);
    spin();

    return 0;
}

File: pub & sub/task1_ws/src/string_pkg/scripts/sub_string.py
#!/usr/bin/python3

import rospy
from std_msgs.msg import String


def listener():
    # create sub node
    rospy.init_node("listener", anonymous=True)
    # create sub object
    sub = rospy.Subscriber("string_topic", String, callback)
    # processes incoming messages until the node is shutdown
    rospy.spin()

def callback(msg):
    rospy.loginfo("Hello, %s", msg.data)

if __name__ == "__main__":
    listener()